File: lua/Ierusalimschy_3ed_book/ch09/ex03.lua
#!/usr/bin/lua52
-- Copyright (C) Khudyashev Ivan, 2018, bahek1983@gmail.com
-- Learning Lua Project
local function dispatch()
    local shList = {
        f1 = coroutine.create(f1),
        f2 = coroutine.create(f2),
        f3 = coroutine.create(f3),
        f4 = coroutine.create(f4),
        f5 = coroutine.create(f5)
    }
    local state, nextF = coroutine.resume(shList["f1"])
    -- print("nextF = ", nextF)
    --print("type(shList[nextF]) = ", type(shList[nextF]))
    while(nextF) do
        state, nextF = coroutine.resume(shList[nextF])
    end
end

function transfer(coroutineName)
    coroutine.yield(coroutineName)
end

function f1()
    print "f1(): I'm function f1"
    print "f1(): call f2()"
    transfer "f2"
    print "f1(): Again here"
    print "f1(): call f5()"
    transfer "f5"
    print "f1(): ending work"
end

function f2()
    print "f2(): I'm f2"
    transfer "f3"
    print "f2(): ending work"
end

function f3()
    print "f3(): I'm f3"
    transfer "f1"
end

function f4()
end

function f5()
    print "f5(): I'm f5"
    transfer "f1"
end

dispatch()

File: c/Prata_languageC/ch06/ex03.c
// Copyright (C) Khudyashev Ivan, 2018-2019, bahek1983@gmail.com
#include <stdio.h>

int main(void)
{
    const int row = 6;
    const char printSymb = 'A';
    printf("Start Program\n\n");
    // program here
    for(int i = 1; i <= row; i++) {
        for(int j = 1; j <= i; j++)
            printf("%c", printSymb + row - j);
        printf("%c", '\n');
    }

    printf("\nFinish Program!\n");
    return 0;
}
